### ISSUE TITLE
Provide a way to deploy actions already deployed.

### ISSUE THREAD

**Issue by lionelvillard** [openwhisk/openwhisk-wskdeploy#208]
Currently `wskdeploy` does not support updating deployed actions. I think this should be the default behavior and eventually we could add a flag (like maybe `--safe` or `--create`) to prevent this default behavior

**Comment by pritidesai** [comment]
_wskdeploy_ errors out saying "Got error creating action with error message: resource already exists (code 5812632) and error code: 153" if you try to deploy an action which already exists. Ideal behavior would be to delete an existing action and redeploy it based on the current manifest and deployment configuration. 

It produces same error for rules and triggers if you try to deploy already existing rules and triggers.

We should update wskdeploy to be consistent with action, rules, and triggers, and support their deployment when they already exist.

**Comment by lionelvillard** [comment]
yes I agree. 577725355b3cae319f3efa1d0f5b69eef8343449 is not good enough.

**Comment by paulcastro** [comment]
We're passing along error messages we getting from the openwhisk-client-go (aka go-whisk) package.  I agree the error messages should be more descriptive than `5777253 `

**Comment by mrutkows** [comment]
This may be related to David work (a week or so ago) to verify what has been installed.  This would be a pre-cursor to "check" a deployment.  Using -verify to verify manifest is reflected in the Cloud.

**Comment by mrutkows** [comment]
Would prefer having the "deployer" confirm (as part of the command) the permission to "update" the deployed application. That is provide, --update as an optional parm./flag.

**Comment by mrutkows** [comment]
Need to open larger epic to track "udpate" feature @daisy-ycguo

**Comment by daisy-ycguo** [comment]
I verified by a test case, deploying another time after it has already been deployed. The current behavior is to `update` of the existing assets. 

I also looked into our source code, we set `overwrite` to be `true` as default. ( Refer to this line of code [here](https://github.com/apache/incubator-openwhisk-wskdeploy/blob/master/deployers/servicedeployer.go#L616)). Do we want to use an option `--update` to change this value ? @mrutkows

**Comment by mrutkows** [comment]
@daisy this is tied to the "user" version that is still to be implemented.  We will need to eventually compare the deployed (user) version of a package with new version in a manifest and have a new flag to allow an update (and prompt interactively as well).

**Comment by pritidesai** [comment]
@mrutkows is this still pending after adding managed deployments? Managed deployment addresses updates in manifest file.
